Vector.cpp

```diff
@@ -0,0 +1,238 @@
+#include <cassert>
+#include <iostream>
+
+class Custom_Vector
+{
+private:
+    size_t _size{};
+    size_t _capacity{};
+    int *buffer;
+
+public:
+    Custom_Vector()
+        : _size(0),
+          _capacity(0),
+          buffer(nullptr) {}
+
+    explicit Custom_Vector(size_t size)
+        : _size(size),
+          _capacity(size),
+          buffer(new int[_capacity])
+    {
+        std::fill_n(buffer, _size, int());
+    }
+
+    Custom_Vector(const std::initializer_list<int> list)
+        : Custom_Vector(list.size())
+    {
+        std::copy(list.begin(), list.end(), buffer);
+    }
+
+    Custom_Vector(const Custom_Vector &other)
+        : Custom_Vector(other._size)
+    {
+        std::copy(other.buffer, other.buffer + other._size, buffer);
+    }
+
+    Custom_Vector(Custom_Vector &&other)
+        : Custom_Vector()
```

Comment:
Similar thing here. The current implementation initializes `size` and `capacity` with zeros and then swaps. You can initialize them directly with the proper values. You need only buffer swapping because it is heap-allocated memory.